### PR TITLE
added factory for preprocessor creation.

### PR DESCRIPTION
**Pull request by sasurobert**
factory for preprocessor creation.

### CODE REVIEW

File: process/factory/shard/preProcessorsContainerFactory.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		txProcessor:       txProcessor,
+		accounts:          accounts,
+		scProcessor:       scProcessor,
+		scResultProcessor: scResultProcessor,
+		interProcessor:    interProcessor,
+		requestHandler:    requestHandler,
+	}, nil
+}
+
+func (ppcm *preProcessorsContainerFactory) Create() (process.PreProcessorsContainer, error) {
```
Comment (ccorcoveanu) [inline process/factory/shard/preProcessorsContainerFactory.go]: Comment missing on exported method

File: process/factory/shard/preProcessorsContainerFactory.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	container := containers.NewPreProcessorsContainer()
+
+	key, preproc, err := ppcm.createTxPreProcessor()
```
Comment (ccorcoveanu) [inline process/factory/shard/preProcessorsContainerFactory.go]: Why is key returned from this function? It is not actually used there, so it's not useful. You can just simply return two values (preproc and err) and do `err = container.Add(block.TxBlock, preproc)`

File: process/factory/shard/preProcessorsContainerFactors_test.go [hunk 3]
```diff
@@ -0,0 +1,400 @@
+package shard
```
Comment (ccorcoveanu) [inline process/factory/shard/preProcessorsContainerFactors_test.go]: Typo in the file's name (Factors)

File: process/factory/shard/preProcessorsContainerFactory.go (hunk 1 above)
Comment (sasurobert) [reply process/factory/shard/preProcessorsContainerFactory.go]: done

File: process/factory/shard/preProcessorsContainerFactory.go (hunk 2 above)
Comment (sasurobert) [reply process/factory/shard/preProcessorsContainerFactory.go]: done

File: process/factory/shard/preProcessorsContainerFactors_test.go (hunk 3 above)
Comment (sasurobert) [reply process/factory/shard/preProcessorsContainerFactors_test.go]: done

**Pull request merged**